File: tests/nodejs/cases/case_259_shadowed_sink_local_05.js
/**
 * Task case 259: shadowed_sink_local_05 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case259ShadowedSinkLocal05(input) {
  const taskName = 'shadowed_sink_local_05';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const retainedCache = new Map();
  retainedCache.set('tmp', payload);
  // SAFE: local shadow hides global sink symbol.
  return payload.checksum;
}

module.exports = { case259ShadowedSinkLocal05 };

File: tests/go/cases/case_259_shadowed_sink_local_05.go
package escape_tests

var retainedCase259 = []map[string]string{}

func Case259ShadowedSinkLocal05(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "shadowed_sink_local_05",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "shadowed_sink_local_05:" + raw,
	}
	retainedCase := []map[string]string{}
	retainedCase = append(retainedCase, payload)
	// SAFE: local shadow variable only.
	return payload["checksum"]
}
